### ISSUE TITLE
Assist Microphone Addon: ALSA lib underrun occurred (inside a VM)

### ISSUE THREAD

**Issue by ms1design** [rhasspy/hassio-addons#26]
Hi, today I was playing with **Assist Microphone Add-on** for HA. Unfortunately I get only to the point where i get this error in **Assist Microphone** Logs.

Edit: It's worth to mention, that a standard **rhasspy Add-on** is working perfectly fine with the same setup.

### My system
- Proxmox VM with HA OS (pass through sound card & Anker PowerConf S330)
- Home Assistant 2023.10.2
- Supervisor 2023.10.0
- Operating System 10.5
- UI 20231005.0 - latest

### Connected Speakers/Mics/Sound Cards
- Anker PowerConf S330 (USB) - chosen in **Assist Microphone Add-on** configuration
- External SoundCard used by **Mopidy Add-on** (later I removed it without any impact on issue)
- Intel HDA sound card pass through to VM

### **Assist Microphone** Logs
```
...
ALSA lib pcm.c:8570:(snd_pcm_recover) underrun occurred
ALSA lib pcm.c:8570:(snd_pcm_recover) underrun occurred
ALSA lib pcm.c:8570:(snd_pcm_recover) underrun occurred
...
```

### `ha audio info`
```
[core-ssh ~]$ ha audio info
audio:
  application:
  - addon: 47701997_assist_microphone
    index: 193
    mute: false
    name: ALSA plug-in [python3.11]
    stream_index: 0
    stream_type: output
    volume: 1
  - addon: 47701997_assist_microphone
    index: 1
    mute: false
    name: ALSA plug-in [python3.11]
    stream_index: 1
    stream_type: input
    volume: 1
  card:
  - driver: module-alsa-card.c
    index: 0
    name: alsa_card.usb-ANKER_Anker_PowerConf_S330_ACCUDP1C49600164-00
    profiles:
    - active: false
      description: Analog Stereo Input
      name: input:analog-stereo
    - active: false
      description: Digital Stereo (IEC958) Input
      name: input:iec958-stereo
    - active: false
      description: Analog Stereo Output
      name: output:analog-stereo
    - active: true
      description: Analog Stereo Duplex
      name: output:analog-stereo+input:analog-stereo
    - active: false
      description: Analog Stereo Output + Digital Stereo (IEC958) Input
      name: output:analog-stereo+input:iec958-stereo
    - active: false
      description: Digital Stereo (IEC958) Output
      name: output:iec958-stereo
    - active: false
      description: Digital Stereo (IEC958) Output + Analog Stereo Input
      name: output:iec958-stereo+input:analog-stereo
    - active: false
      description: Digital Stereo Duplex (IEC958)
      name: output:iec958-stereo+input:iec958-stereo
    - active: false
      description: "Off"
      name: "off"
  - driver: module-alsa-card.c
    index: 1
    name: alsa_card.pci-0000_08_0c.0
    profiles:
    - active: false
      description: Analog Stereo Input
      name: input:analog-stereo
    - active: false
      description: Analog Stereo Output
      name: output:analog-stereo
    - active: true
      description: Analog Stereo Duplex
      name: output:analog-stereo+input:analog-stereo
    - active: false
      description: "Off"
      name: "off"
  input:
  - applications:
    - addon: 47701997_assist_microphone
      index: 1
      mute: false
      name: ALSA plug-in [python3.11]
      stream_index: 1
      stream_type: input
      volume: 1
    card: 0
    default: true
    description: Anker PowerConf S330 Analog Stereo
    index: 1
    mute: false
    name: alsa_input.usb-ANKER_Anker_PowerConf_S330_ACCUDP1C49600164-00.analog-stereo
    volume: 1
  - applications: []
    card: 1
    default: false
    description: HDA Intel Analog Stereo
    index: 3
    mute: true
    name: alsa_input.pci-0000_08_0c.0.analog-stereo
    volume: 0.058441162109375
  output:
  - applications:
    - addon: 47701997_assist_microphone
      index: 193
      mute: false
      name: ALSA plug-in [python3.11]
      stream_index: 0
      stream_type: output
      volume: 1
    card: 0
    default: true
    description: Anker PowerConf S330 Analog Stereo
    index: 0
    mute: false
    name: alsa_output.usb-ANKER_Anker_PowerConf_S330_ACCUDP1C49600164-00.analog-stereo
    volume: 1
  - applications: []
    card: 1
    default: false
    description: HDA Intel Analog Stereo
    index: 1
    mute: false
    name: alsa_output.pci-0000_08_0c.0.analog-stereo
    volume: 1
host: 172.30.32.4
update_available: false
version: 2023.06.0
version_latest: 2023.06.0
```

### `ha audio log`
```
[core-ssh ~]$ ha audio log
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/alsa-mixer.sh
[17:38:34] INFO: Adjust ALSA mixer settings for /dev/snd/controlC0
[17:38:35] INFO: Adjust ALSA mixer settings for /dev/snd/controlC1
[17:38:35] INFO: Adjust ALSA mixer settings for /dev/snd/controlC2
cont-init: info: /etc/cont-init.d/alsa-mixer.sh exited 0
cont-init: info: running /etc/cont-init.d/filesystem.sh
cont-init: info: /etc/cont-init.d/filesystem.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service legacy-services: starting
services-up: info: copying legacy longrun alsa (no readiness notification)
services-up: info: copying legacy longrun pulseaudio (no readiness notification)
s6-rc: info: service legacy-services successfully started
W: [pulseaudio] main.c: Running in system mode, but --disallow-module-loading not set.
W: [pulseaudio] main.c: Home directory of user 'root' is not '/var/run/pulse', ignoring.
W: [pulseaudio] caps.c: Normally all extra capabilities would be dropped now, but that's impossible because PulseAudio was built without capabilities support.
W: [pulseaudio] main.c: OK, so you are running PA in system mode. Please make sure that you actually do want to do that.
W: [pulseaudio] main.c: Please read http://www.freedesktop.org/wiki/Software/PulseAudio/Documentation/User/WhatIsWrongWithSystemWide/ for an explanation why system mode is usually a bad idea.
W: [pulseaudio] module-udev-detect.c: Failed to open /proc/asound/card0: No such file or directory
E: [pulseaudio] module-alsa-card.c: Failed to find a working profile.
E: [pulseaudio] module.c: Failed to load module "module-alsa-card" (argument: "device_id="0" name="pci-0000_00_1b.0" card_name="alsa_card.pci-0000_00_1b.0" namereg_fail=false tsched=yes fixed_latency_range=no ignore_dB=no deferred_volume=yes use_ucm=yes avoid_resampling=no card_properties="module-udev-detect.discovered=1""): initialization failed.
W: [pulseaudio] module-udev-detect.c: Failed to open /proc/asound/card2: No such file or directory
N: [pulseaudio] alsa-util.c: Disabling timer-based scheduling because running inside a VM.
N: [pulseaudio] alsa-util.c: Disabling timer-based scheduling because running inside a VM.
W: [pulseaudio] module-udev-detect.c: Failed to open /proc/asound/card1: No such file or directory
N: [pulseaudio] alsa-util.c: Disabling timer-based scheduling because running inside a VM.
N: [pulseaudio] alsa-util.c: Disabling timer-based scheduling because running inside a VM.
E: [pulseaudio] module-rescue-streams.c: module-rescue-stream is obsolete and should no longer be loaded. Please remove it from your configuration.
E: [alsa-sink-USB Audio] alsa-sink.c: ALSA woke us up to write new data to the device, but there was actually nothing to write.
E: [alsa-sink-USB Audio] alsa-sink.c: Most likely this is a bug in the ALSA driver 'snd_usb_audio'. Please report this issue to the ALSA developers.
E: [alsa-sink-USB Audio] alsa-sink.c: We were woken up with POLLOUT set -- however a subsequent snd_pcm_avail() returned 0 or another value < min_avail.
E: [alsa-source-USB Audio] alsa-source.c: ALSA woke us up to read new data from the device, but there was actually nothing to read.
E: [alsa-source-USB Audio] alsa-source.c: Most likely this is a bug in the ALSA driver 'snd_usb_audio'. Please report this issue to the ALSA developers.
E: [alsa-source-USB Audio] alsa-source.c: We were woken up with POLLIN set -- however a subsequent snd_pcm_avail() returned 0 or another value < min_avail.
```

**Comment by Lampyridae1984** [comment]
+1

Using Anker PowerConf S330 Analog Stereo (latest firmware)

Home Assistant Yellow
No external sound card.

Home Assistant 2023.10.2
Supervisor 2023.10.0
Operating System 10.5

**Comment by ms1design** [comment]
Resolved in 4731121de02bff300f2be5e3c532afa2e86e22a5
